pkg/queue/readiness/probe.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package readiness
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/knative/serving/pkg/queue/health"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultProbeTimeout = 100 * time.Millisecond
```

Comment:
This is defined here and in `main.go`. Is this supposed to stay in sync? If so we should make this public and share it between the two places.

Reply:
The probe timeout in `probe.go` and `main.go` are used for different things that don't necessary need to sync.

`defaultProbeTimeout` in `main.go` is used in `probeUserContainer` which is used as a signal to the activator that the pod is up and ready to accept traffic. If we want to keep using the pre-existing, hard-coded TCP connect as the activator's test for readiness, it seems like the constants are two similar, but different, things.

Reply:
Should the activator's test be the same as the one that's set off from kubelet?

Reply:
@vagababov @mattmoor Any thoughts here?